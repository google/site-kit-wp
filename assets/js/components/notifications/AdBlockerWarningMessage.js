/**
 * AdBlockerWarningMessage component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { sprintf, __ } from '@wordpress/i18n';
import { createInterpolateElement } from '@wordpress/element';

/**
 * Internal dependencies
 */
import Link from '../Link';
import WarningNotice from '../WarningNotice';
import ExternalIcon from '../../../svg/icons/external-rounded.svg';

export default function AdBlockerWarningMessage( {
	className = '',
	getHelpLink = '',
	warningMessage = null,
} ) {
	if ( ! warningMessage ) {
		return null;
	}

	return (
		<WarningNotice className={ className }>
			{ createInterpolateElement(
				sprintf(
					/* translators: 1: The warning message. 2: "Get help" text. */
					__( '%1$s. <Link>%2$s</Link>', 'google-site-kit' ),
					warningMessage,
					__( 'Get help', 'google-site-kit' )
				),
				{
					Link: (
						<Link
							href={ getHelpLink }
							external
							hideExternalIndicator
							trailingIcon={
								<ExternalIcon width={ 15 } height={ 15 } />
							}
						/>
					),
				}
			) }
		</WarningNotice>
	);
}

AdBlockerWarningMessage.propTypes = {
	className: PropTypes.string,
	getHelpLink: PropTypes.string,
	warningMessage: PropTypes.string,
};
