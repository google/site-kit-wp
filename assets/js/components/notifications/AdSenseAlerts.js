/**
 * AdSenseAlerts component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { Fragment } from '@wordpress/element';
import { __, _x } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
import { MODULES_ADSENSE } from '../../modules/adsense/datastore/constants';
import NotificationAlertSVG from '../../../svg/graphics/notification-alert.svg';
import BannerNotification from '../notifications/BannerNotification';

function AdSenseAlerts() {
	const adSenseModuleConnected = useSelect( ( select ) =>
		select( CORE_MODULES ).isModuleConnected( 'adsense' )
	);
	const accountID = useSelect( ( select ) =>
		select( MODULES_ADSENSE ).getAccountID()
	);
	const notifications = useSelect( ( select ) =>
		select( MODULES_ADSENSE ).getNotifications()
	);

	if (
		! adSenseModuleConnected ||
		! accountID ||
		notifications === undefined
	) {
		return null;
	}

	return (
		<Fragment>
			{ notifications.map(
				( {
					id,
					title,
					description,
					format,
					ctaURL,
					ctaLabel,
					ctaTarget,
					severity,
					isDismissable,
				} ) => (
					<BannerNotification
						className="googlesitekit-adsense-alert"
						key={ id }
						id={ id }
						title={ title || '' }
						description={ description }
						WinImageSVG={ NotificationAlertSVG }
						format={ format || 'small' }
						ctaLink={ ctaURL }
						ctaLabel={ ctaLabel }
						ctaTarget={ ctaTarget }
						type={ severity }
						dismiss={ __( 'OK, Got it!', 'google-site-kit' ) }
						isDismissible={ isDismissable || true }
						logo
						module="adsense"
						moduleName={ _x(
							'AdSense',
							'Service name',
							'google-site-kit'
						) }
						dismissExpires={ 0 }
						showOnce={ false }
					/>
				)
			) }
		</Fragment>
	);
}

export default AdSenseAlerts;
