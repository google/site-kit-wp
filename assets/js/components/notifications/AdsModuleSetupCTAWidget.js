/**
 * AdsModuleSetupCTAWidget component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { compose } from '@wordpress/compose';
import { useCallback } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import { Button } from 'googlesitekit-components';
import {
	BREAKPOINT_SMALL,
	BREAKPOINT_TABLET,
	useBreakpoint,
} from '../../hooks/useBreakpoint';
import { ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY } from '../../modules/ads/datastore/constants';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { WEEK_IN_SECONDS } from '../../util';
import { Cell, Grid, Row } from '../../material-components';
import whenInactive from '../../util/when-inactive';
import { withWidgetComponentProps } from '../../googlesitekit/widgets/util';
import useActivateModuleCallback from '../../hooks/useActivateModuleCallback';
import AdsSetupSVG from '../../../svg/graphics/ads-setup.svg';
import AdsSetupTabletSVG from '../../../svg/graphics/ads-setup-tablet.svg';
import AdsSetupMobileSVG from '../../../svg/graphics/ads-setup-mobile.svg';

function AdsModuleSetupCTAWidget( { WidgetNull, Widget } ) {
	const breakpoint = useBreakpoint();
	const isMobileBreakpoint = breakpoint === BREAKPOINT_SMALL;
	const isTabletBreakpoint = breakpoint === BREAKPOINT_TABLET;

	const onSetupCallback = useActivateModuleCallback( 'ads' );

	const isDismissed = useSelect( ( select ) =>
		select( CORE_USER ).isPromptDismissed(
			ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY
		)
	);
	const dismissCount = useSelect( ( select ) =>
		select( CORE_USER ).getPromptDismissCount(
			ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY
		)
	);

	const { dismissPrompt } = useDispatch( CORE_USER );

	const onDismiss = useCallback( async () => {
		if ( dismissCount < 1 ) {
			const twoWeeksInSeconds = WEEK_IN_SECONDS * 2;

			await dismissPrompt( ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY, {
				expiresInSeconds: twoWeeksInSeconds,
			} );
		} else {
			// For the second dismissal, dismiss permanently.
			await dismissPrompt( ADS_MODULE_SETUP_BANNER_PROMPT_DISMISSED_KEY );
		}
	}, [ dismissCount, dismissPrompt ] );

	if ( dismissCount > 1 || isDismissed || isDismissed === undefined ) {
		return <WidgetNull />;
	}

	return (
		<div className="googlesitekit-widget-context">
			<Grid className="googlesitekit-widget-area">
				<Row>
					<Cell size={ 12 }>
						<Widget
							noPadding
							className="googlesitekit-setup-cta-banner googlesitekit-ads-setup-cta-widget"
						>
							<Grid collapsed>
								<Row>
									<Cell
										smSize={ 12 }
										mdSize={ 8 }
										lgSize={ 6 }
										className="googlesitekit-setup-cta-banner__primary-cell"
									>
										<h3 className="googlesitekit-setup-cta-banner__title">
											{ __(
												'Get better quality leads and enhance conversions with Ads',
												'google-site-kit'
											) }
										</h3>
										<div className="googlesitekit-setup-cta-banner__description">
											<p>
												{ __(
													'Help drive sales, leads, or site traffic by getting your business in front of people who are actively searching Google for products or services you offer.',
													'google-site-kit'
												) }
											</p>
										</div>

										<div className="googlesitekit-setup-cta-banner__actions-wrapper">
											<Button
												className="googlesitekit-key-metrics-cta-button"
												onClick={ onSetupCallback }
											>
												{ __(
													'Set up Ads',
													'google-site-kit'
												) }
											</Button>
											<Button
												tertiary
												onClick={ onDismiss }
											>
												{ dismissCount < 1
													? __(
															'Maybe later',
															'google-site-kit'
													  )
													: __(
															'Donâ€™t show again',
															'google-site-kit'
													  ) }
											</Button>
										</div>
									</Cell>
									{ ! isMobileBreakpoint &&
										! isTabletBreakpoint && (
											<Cell
												alignBottom
												className="googlesitekit-setup-cta-banner__svg-wrapper"
												mdSize={ 8 }
												lgSize={ 6 }
											>
												<AdsSetupSVG />
											</Cell>
										) }
									{ isTabletBreakpoint && (
										<Cell
											className="googlesitekit-setup-cta-banner__svg-wrapper"
											mdSize={ 8 }
										>
											<AdsSetupTabletSVG />
										</Cell>
									) }
									{ isMobileBreakpoint && (
										<Cell
											alignBottom
											className="googlesitekit-setup-cta-banner__svg-wrapper"
											smSize={ 12 }
										>
											<AdsSetupMobileSVG />
										</Cell>
									) }
								</Row>
							</Grid>
						</Widget>
					</Cell>
				</Row>
			</Grid>
		</div>
	);
}

AdsModuleSetupCTAWidget.propTypes = {
	Widget: PropTypes.elementType.isRequired,
	WidgetNull: PropTypes.elementType.isRequired,
};

export default compose(
	whenInactive( { moduleName: 'ads' } ),
	withWidgetComponentProps( 'ads-setup-cta' )
)( AdsModuleSetupCTAWidget );
