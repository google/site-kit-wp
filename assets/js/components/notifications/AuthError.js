/**
 * AuthError component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import NotificationError from '../../googlesitekit/notifications/components/layout/NotificationError';
import CTALink from '../../googlesitekit/notifications/components/common/CTALink';
import Description from '../../googlesitekit/notifications/components/common/Description';

export default function AuthError( { id, Notification } ) {
	const error = useSelect( ( select ) => select( CORE_USER ).getAuthError() );

	return (
		<Notification className="googlesitekit-publisher-win googlesitekit-publisher-win--win-error">
			<NotificationError
				title={ __(
					'Site Kit canâ€™t access necessary data',
					'google-site-kit'
				) }
				description={ <Description text={ error.message } /> }
				actions={
					<CTALink
						id={ id }
						ctaLabel={ __(
							'Redo the plugin setup',
							'google-site-kit'
						) }
						ctaLink={ error.data.reconnectURL }
					/>
				}
			/>
		</Notification>
	);
}
