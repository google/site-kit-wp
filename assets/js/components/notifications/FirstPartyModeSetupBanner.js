/**
 * FirstPartyModeSetupBanner component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { Fragment, useCallback, useEffect } from '@wordpress/element';
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import { useDispatch, useSelect } from 'googlesitekit-data';
import {
	useShowTooltip,
	useTooltipState,
	AdminMenuTooltip,
} from '../AdminMenuTooltip';
import {
	CORE_NOTIFICATIONS,
	NOTIFICATION_GROUPS,
} from '../../googlesitekit/notifications/datastore/constants';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
import Description from '../../googlesitekit/notifications/components/common/Description';
import LearnMoreLink from '../../googlesitekit/notifications/components/common/LearnMoreLink';
import NotificationWithSVG from '../../googlesitekit/notifications/components/layout/NotificationWithSVG';
import ActionsCTALinkDismiss from '../../googlesitekit/notifications/components/common/ActionsCTALinkDismiss';
import FPMSetupCTASVGDesktop from '../../../svg/graphics/first-party-mode-setup-banner-desktop.svg';
import FPMSetupCTASVGTablet from '../../../svg/graphics/first-party-mode-setup-banner-tablet.svg';
import FPMSetupCTASVGMobile from '../../../svg/graphics/first-party-mode-setup-banner-mobile.svg';
import {
	BREAKPOINT_SMALL,
	BREAKPOINT_TABLET,
	useBreakpoint,
} from '../../hooks/useBreakpoint';
import useViewContext from '../../hooks/useViewContext';
import { DAY_IN_SECONDS, trackEvent } from '../../util';

export const FPM_SHOW_SETUP_SUCCESS_NOTIFICATION =
	'fpm-show-setup-success-notification';

export default function FirstPartyModeSetupBanner( { id, Notification } ) {
	const breakpoint = useBreakpoint();
	const viewContext = useViewContext();

	const { setFirstPartyModeEnabled, saveFirstPartyModeSettings } =
		useDispatch( CORE_SITE );

	const showTooltip = useShowTooltip( id );

	const { isTooltipVisible } = useTooltipState( id );

	const usingProxy = useSelect( ( select ) =>
		select( CORE_SITE ).isUsingProxy()
	);

	const isItemDismissed = useSelect( ( select ) =>
		select( CORE_NOTIFICATIONS ).isNotificationDismissed( id )
	);

	const { invalidateResolution } = useDispatch( CORE_NOTIFICATIONS );

	const isDismissing = useSelect( ( select ) =>
		select( CORE_USER ).isDismissingItem( id )
	);

	const { setValue } = useDispatch( CORE_UI );

	const learnMoreURL = useSelect( ( select ) => {
		return select( CORE_SITE ).getDocumentationLinkURL(
			'first-party-mode-introduction'
		);
	} );

	useEffect( () => {
		if ( isTooltipVisible ) {
			trackEvent( `${ viewContext }_fpm-setup-cta`, 'tooltip_view' );
		}
	}, [ isTooltipVisible, viewContext ] );

	const onCTAClick = async () => {
		setFirstPartyModeEnabled( true );
		const { error } = await saveFirstPartyModeSettings();

		if ( error ) {
			return { error };
		}

		setValue( FPM_SHOW_SETUP_SUCCESS_NOTIFICATION, true );
		invalidateResolution( 'getQueuedNotifications', [
			viewContext,
			NOTIFICATION_GROUPS.DEFAULT,
		] );
	};

	const { triggerSurvey } = useDispatch( CORE_USER );
	const handleView = useCallback( () => {
		if ( usingProxy ) {
			triggerSurvey( 'view_fpm_setup_cta', { ttl: DAY_IN_SECONDS } );
		}
	}, [ triggerSurvey, usingProxy ] );

	const onDismiss = () => {
		showTooltip();
	};

	if ( isTooltipVisible ) {
		return (
			<Fragment>
				<AdminMenuTooltip
					title=""
					content={ __(
						'You can always enable First-party mode from Settings later',
						'google-site-kit'
					) }
					dismissLabel={ __( 'Got it', 'google-site-kit' ) }
					onDismiss={ () => {
						trackEvent(
							`${ viewContext }_fpm-setup-cta`,
							'tooltip_dismiss'
						);
					} }
					tooltipStateKey={ id }
				/>
			</Fragment>
		);
	}

	if ( isItemDismissed || isDismissing ) {
		return null;
	}

	const breakpointSVGMap = {
		[ BREAKPOINT_SMALL ]: FPMSetupCTASVGMobile,
		[ BREAKPOINT_TABLET ]: FPMSetupCTASVGTablet,
	};

	return (
		<Notification onView={ handleView }>
			<NotificationWithSVG
				id={ id }
				title={ __(
					'Get more comprehensive stats by collecting metrics via your own site',
					'google-site-kit'
				) }
				description={
					<Description
						text={ __(
							'Enable First-party mode (<em>beta</em>) to send measurement through your own domain - this helps improve the quality and completeness of Analytics or Ads metrics.',
							'google-site-kit'
						) }
						learnMoreLink={
							<LearnMoreLink
								id={ id }
								label={ __( 'Learn more', 'google-site-kit' ) }
								url={ learnMoreURL }
							/>
						}
					/>
				}
				actions={
					<ActionsCTALinkDismiss
						id={ id }
						className="googlesitekit-setup-cta-banner__actions-wrapper"
						ctaLabel={ __(
							'Enable First-party mode',
							'google-site-kit'
						) }
						onCTAClick={ onCTAClick }
						ctaDismissOptions={ {
							skipHidingFromQueue: false,
						} }
						dismissLabel={ __( 'Maybe later', 'google-site-kit' ) }
						onDismiss={ onDismiss }
						dismissOptions={ {
							skipHidingFromQueue: true,
						} }
					/>
				}
				SVG={ breakpointSVGMap[ breakpoint ] || FPMSetupCTASVGDesktop }
			/>
		</Notification>
	);
}
