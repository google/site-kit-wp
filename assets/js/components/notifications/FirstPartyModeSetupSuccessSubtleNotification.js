/**
 * FirstPartyModeSetupSuccessSubtleNotification component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';

/**
 * Internal dependencies
 */
import SubtleNotification from '../../googlesitekit/notifications/components/layout/SubtleNotification';
import Dismiss from '../../googlesitekit/notifications/components/common/Dismiss';

export const FIRST_PARTY_MODE_SETUP_SUCCESS_NOTIFICATION =
	'setup-success-notification-fpm';

export default function FirstPartyModeSetupSuccessSubtleNotification( {
	id,
	Notification,
} ) {
	return (
		<Notification>
			<SubtleNotification
				title={ __(
					'You successfully enabled First-party mode!',
					'google-site-kit'
				) }
				description={ __(
					'You can always disable it in Analytics or Ads settings',
					'google-site-kit'
				) }
				dismissCTA={
					<Dismiss
						id={ id }
						primary={ false }
						dismissLabel={ __( 'Got it', 'google-site-kit' ) }
					/>
				}
			/>
		</Notification>
	);
}

FirstPartyModeSetupSuccessSubtleNotification.propTypes = {
	id: PropTypes.string.isRequired,
	Notification: PropTypes.elementType.isRequired,
};
