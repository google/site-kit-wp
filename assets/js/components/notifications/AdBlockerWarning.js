/**
 * AdSense AdBlockerWarning component.
 *
 * Site Kit by Google, Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import { useSelect } from 'googlesitekit-data';
import { CORE_SITE } from '../../googlesitekit/datastore/site/constants';
import { CORE_MODULES } from '../../googlesitekit/modules/datastore/constants';
import AdBlockerWarningMessage from './AdBlockerWarningMessage';

export default function AdBlockerWarning( { moduleSlug, className } ) {
	const storeName = useSelect( ( select ) =>
		select( CORE_MODULES ).getModuleStoreName( moduleSlug )
	);
	const adBlockerWarningMessage = useSelect( ( select ) =>
		select( storeName )?.getAdBlockerWarningMessage()
	);
	const getHelpLink = useSelect( ( select ) =>
		select( CORE_SITE ).getDocumentationLinkURL(
			`${ moduleSlug }-ad-blocker-detected`
		)
	);

	return (
		<AdBlockerWarningMessage
			className={ className }
			getHelpLink={ getHelpLink }
			warningMessage={ adBlockerWarningMessage }
		/>
	);
}

AdBlockerWarning.propTypes = {
	className: PropTypes.string,
	moduleSlug: PropTypes.string.isRequired,
};
