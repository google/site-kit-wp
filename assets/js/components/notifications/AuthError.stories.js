/**
 * AuthError Component Stories.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import WithRegistrySetup from '../../../../tests/js/WithRegistrySetup';
import { provideUserAuthentication } from '../../../../tests/js/utils';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import AuthError from './AuthError';
import { withNotificationComponentProps } from '../../googlesitekit/notifications/util/component-props';

const NotificationWithComponentProps =
	withNotificationComponentProps( 'auth-error' )( AuthError );

function Template() {
	return <NotificationWithComponentProps />;
}

export const AuthenticationError = Template.bind( {} );
AuthenticationError.storyName = 'AuthError';
AuthenticationError.scenario = {};

export default {
	title: 'Components/Notifications/Banners',
	decorators: [
		( Story ) => {
			const setupRegistry = ( registry ) => {
				provideUserAuthentication( registry );

				const authError = {
					code: 'missing_delegation_consent',
					status: 401,
					message:
						'Looks like your site is not allowed access to Google account data and canâ€™t display stats in the dashboard.',
					data: {
						reconnectURL: 'https://example.com/',
					},
				};

				registry.dispatch( CORE_USER ).setAuthError( authError );
			};

			return (
				<WithRegistrySetup func={ setupRegistry }>
					<Story />
				</WithRegistrySetup>
			);
		},
	],
};
