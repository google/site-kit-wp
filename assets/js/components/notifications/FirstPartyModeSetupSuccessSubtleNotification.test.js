/**
 * FirstPartyModeSetupSuccessSubtleNotification component tests.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import { createTestRegistry, render } from '../../../../tests/js/test-utils';
import { CORE_UI } from '../../googlesitekit/datastore/ui/constants';
import { VIEW_CONTEXT_MAIN_DASHBOARD } from '../../googlesitekit/constants';
import { DEFAULT_NOTIFICATIONS } from '../../googlesitekit/notifications/register-defaults';
import FirstPartyModeSetupSuccessSubtleNotification from './FirstPartyModeSetupSuccessSubtleNotification';
import { withNotificationComponentProps } from '../../googlesitekit/notifications/util/component-props';
import { FPM_SHOW_SETUP_SUCCESS_NOTIFICATION } from './FirstPartyModeSetupBanner';

const NotificationWithComponentProps = withNotificationComponentProps(
	'setup-success-notification-fpm'
)( FirstPartyModeSetupSuccessSubtleNotification );

describe( 'FirstPartyModeSetupSuccessSubtleNotification', () => {
	let registry;

	beforeEach( () => {
		registry = createTestRegistry();
	} );

	it( 'should render correctly', () => {
		const { container, getByText } = render(
			<NotificationWithComponentProps />,
			{ registry }
		);

		expect(
			getByText(
				'You can always disable it in Analytics or Ads settings'
			)
		).toBeInTheDocument();

		expect( container ).toMatchSnapshot();
	} );

	describe( 'checkRequirements', () => {
		const notification =
			DEFAULT_NOTIFICATIONS[ 'setup-success-notification-fpm' ];

		it( 'is active when FPM_SHOW_SETUP_SUCCESS_NOTIFICATION is true', () => {
			registry
				.dispatch( CORE_UI )
				.setValue( FPM_SHOW_SETUP_SUCCESS_NOTIFICATION, true );

			const isActive = notification.checkRequirements(
				registry,
				VIEW_CONTEXT_MAIN_DASHBOARD
			);

			expect( isActive ).toBe( true );
		} );

		it( 'is not active when FPM_SHOW_SETUP_SUCCESS_NOTIFICATION is false', () => {
			registry
				.dispatch( CORE_UI )
				.setValue( FPM_SHOW_SETUP_SUCCESS_NOTIFICATION, false );

			const isActive = notification.checkRequirements(
				registry,
				VIEW_CONTEXT_MAIN_DASHBOARD
			);

			expect( isActive ).toBe( false );
		} );
	} );
} );
