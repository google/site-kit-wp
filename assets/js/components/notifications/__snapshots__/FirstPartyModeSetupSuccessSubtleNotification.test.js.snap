// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstPartyModeSetupSuccessSubtleNotification should render correctly 1`] = `
<div>
  <section
    id="setup-success-notification-fpm"
  >
    <div
      class="mdc-layout-grid"
    >
      <div
        class="mdc-layout-grid__inner"
      >
        <div
          class="googlesitekit-subtle-notification googlesitekit-subtle-notification--success mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-12"
        >
          <div
            class="googlesitekit-subtle-notification__icon"
          >
            <svg />
          </div>
          <div
            class="googlesitekit-subtle-notification__content"
          >
            <p>
              You successfully enabled First-party mode!
            </p>
            <p
              class="googlesitekit-subtle-notification__secondary_description"
            >
              You can always disable it in Analytics or Ads settings
            </p>
          </div>
          <div
            class="googlesitekit-subtle-notification__action"
          >
            <button
              class="mdc-button mdc-button--tertiary"
              target="_self"
            >
              <span
                class="mdc-button__label"
              >
                Got it
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
